clueless/static/js/views.js

```diff
@@ -501,7 +507,11 @@ class ChatPanel extends Panel {
     this.btnSend.onclick = this.handleBtnSendClick.bind(this);
   }
   display(gameState) {
-
+    $(this.chatlog).html('');
+    for (var i = 0; i < gameState.chat_log.length; i++) {
+      var log = gameState.chat_log[i];
+      $(this.chatlog).append('<div class="chatentry">' + log.date.getHours() + ':' + log.date.getMinutes() +' <b>' + log.from + ':</b> ' + log.message + "</div>");
+    }
```

Comment:
Confused by this. Is this recreating the whole chat log each time?

Reply:
It is wiping it clean and rewriting it all every time the `ChatPanel.display()` method is being called. Not the most efficient, but it's an artifact of the way the frontend is designed.

Reply:
If it gets slow, we can probably refactor to make it just append, but it seemed to work fine.

Reply:
Ah yeah I just wasn't quite sure what was happening here. Certainly not the most efficient, but I highly doubt it'll be an issue in our limited scope.